tests/unit/Imports/ProductImportTest.php

```diff
@@ -155,6 +157,194 @@ function (MockInterface $module) {
         $this->assertCount(1, $variableProduct->get_upsell_ids(), '1 upsell IDs should be set');
     }
 
+    public function dataProviderTestGetStockProperties()
+    {
+        $tests = [];
+
+        $tests['simple or configurable assign product in stock, limited, has positive orderable value'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => true,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => 10,
+            ],
+            [
+                'in_stock' => true,
+                'manage_stock' => true,
+                'quantity' => 10,
+            ],
+        ];
+
+        $tests['simple or configurable assign product in stock, limited, has negative orderable value'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => true,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => -10,
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => true,
+                'quantity' => 0,
+            ],
+        ];
+
+        $tests['simple or configurable assign product in stock, limited, has no orderable value'] = [
+            [
+                'type' => 'configurable_assign',
+                'product_stock' => [
+                    'in_stock' => true,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => null,
+            ],
+            [
+                'in_stock' => true,
+                'manage_stock' => true,
+                'quantity' => 5,
+            ],
+        ];
+
+        $tests['simple or configurable assign product out stock, limited, has orderable value'] = [
```

Comment:
that on is a strange one, you got it back from backned?

Reply:
Changed tests.